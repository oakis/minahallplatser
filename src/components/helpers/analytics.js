import analytics from '@react-native-firebase/analytics';

const convertSpaces = str => str.replace(/ /g, '_');

export const track = (title, props) => {
  if (props) {
    window.log(`Firebase analytics track with props: ${title}`, props);
    analytics().logEvent(convertSpaces(title), props);
  } else {
    window.log(`Firebase analytics track: ${title}`);
    analytics().logEvent(convertSpaces(title), {});
  }
};
