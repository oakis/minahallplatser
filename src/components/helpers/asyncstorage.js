import { AsyncStorage } from 'react-native';

export const generateUid = () => {
    return `${s4()}${s4()}-${s4()}-${s4()}-${s4()}-${s4()}${s4()}${s4()}`;
};

function s4() {
    return Math.floor((1 + Math.random()) * 0x10000).toString(16).substring(1);
}

/**
 * @param {string} item 
 * @returns {Promise} Loaded storage
 */
export const getStorage = (item) => {
    return new Promise((resolve, reject) => {
        AsyncStorage.getItem(item)
        .then((json) => {
            const data = json === null ? {} : JSON.parse(json);
            window.log(`getStorage() ${item}: OK`, data);
            resolve(data);
        })
        .catch((e) => {
            window.log(`getStorage() ${item}: FAIL`, e);
            reject(e);
        });
    });
};

/**
 * @param {string} item 
 * @param {JSON} json 
 * @returns {Promise} -
 */
export const setStorage = (item, json) => {
    return new Promise((resolve, reject) => {
        AsyncStorage.setItem(item, JSON.stringify(json))
        .then(() => {
            window.log(`setStorage() ${item}: OK`);
            resolve();
        })
        .catch((e) => {
            window.log(`setStorage() ${item}: FAIL`, e);
            reject(e);
        });
    });
};
