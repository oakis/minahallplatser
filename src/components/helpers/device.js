import { Platform } from 'react-native';

export const isAndroid = () => {
    return Platform.OS === 'android';
};

export const isIOS = () => {
    return Platform.OS === 'ios';
};
