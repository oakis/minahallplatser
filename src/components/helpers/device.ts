import {Platform} from 'react-native';
import {
  getModel,
  getSystemVersion,
  getVersion,
  getUniqueId,
} from 'react-native-device-info';

export const isAndroid = (): boolean => Platform.OS === 'android';

export const isIOS = (): boolean => Platform.OS === 'ios';

export const getDeviceModel = (): string => getModel();

export const getOsVersion = (): string => getSystemVersion();

export const getAppVersion = (): string => getVersion();

export const deviceId = (): string => getUniqueId();
