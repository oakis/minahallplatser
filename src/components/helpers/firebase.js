import fetch from 'react-native-cancelable-fetch';
import { firebaseFunctionsUrl } from '../../Server';
import { handleJsonFetch } from './';

export const updateDeparturesCount = (count) => {
    if (__DEV__) {
        return;
    }
    const url = `${firebaseFunctionsUrl}/addDeparturesCount?count=${count}`;
    fetch(url, {}, 'updateDeparturesCount')
    .finally(handleJsonFetch)
    .then(({ message }) => {
        window.log(`updateDeparturesCount(): OK (${message})`);
    })
    .catch((err) => {
        window.log(`updateDeparturesCount(): FAILED' ${message})`);
        new Error(err);
    });
};

export const getDeparturesCount = () => {
    const url = `${firebaseFunctionsUrl}/getDeparturesCount`;
    return fetch(url, {}, 'getDeparturesCount')
    .finally(handleJsonFetch)
    .then(({ departuresCount }) => {
        window.log('getDeparturesCount(): OK', departuresCount);
        return departuresCount;
    })
    .catch((err) => {
        window.log('getDeparturesCount(): FAILED', err);
        new Error(err);
    });
};

export const updateStopsCount = () => {
    if (__DEV__) {
        return;
    }
    const url = `${firebaseFunctionsUrl}/addStopsCount`;
    fetch(url, {}, 'updateStopsCount')
    .finally(handleJsonFetch)
    .then(({ message }) => {
        window.log('updateStopsCount(): OK', message);
    })
    .catch((err) => {
        window.log('updateStopsCount(): FAILED', message);
        new Error(err);
    });
};

export const getStopsCount = () => {
    const url = `${firebaseFunctionsUrl}/getStopsCount`;
    return fetch(url, {}, 'getStopsCount')
    .finally(handleJsonFetch)
    .then(({ stopsCount }) => {
        window.log('getStopsCount(): OK', stopsCount);
        return stopsCount;
    })
    .catch((err) => {
        window.log('getStopsCount(): FAILED', err);
        new Error(err);
    });
};

export const incrementStopsOpened = (user, stopId) => {
    const url = `${firebaseFunctionsUrl}/incrementStopsOpen?user=${user}&stopId=${stopId}`;
    fetch(url, {}, 'incrementStopsOpened')
    .finally(handleJsonFetch)
    .then(({ message }) => {
        window.log('incrementStopsOpened(): OK', message);
    })
    .catch((err) => {
        window.log('incrementStopsOpened(): FAILED', err);
        new Error(err);
    });
}