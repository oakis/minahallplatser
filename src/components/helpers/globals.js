export const globals = {
    shouldExitApp: false,
    didLogout: false,
    isLoggingIn: false
};
