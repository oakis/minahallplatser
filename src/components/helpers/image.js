import {Dimensions} from 'react-native';

export const calcImageSize = (percent, margin) => {
  const dimensions = Dimensions.get('window');
  const imageWidth = dimensions.width * percent - margin;
  const imageHeight = Math.round((imageWidth * 9) / 16);
  return {imageHeight, imageWidth};
};
