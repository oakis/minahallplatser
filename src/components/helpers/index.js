export * from './device';
export * from './message';
export * from './network';
export * from './token';
export * from './firebase';
export * from './analytics';
export * from './globals';
export * from './image';

export const isObject = (obj) => {
    return Object.prototype.toString.call(obj) === '[object Object]';
};
