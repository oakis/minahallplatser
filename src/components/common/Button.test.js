import React from 'react';
import { shallow } from 'enzyme';
import { isAndroid } from '../helpers';
import { Button } from './';

it('should match snapshot on Android', () => {
    isAndroid.mockImplementationOnce(() => true);
    const wrapper = shallow(<Button />);
    expect(wrapper).toMatchSnapshot();
});

it('should match snapshot on iOS', () => {
    isAndroid.mockImplementationOnce(() => false);
    const wrapper = shallow(<Button />);
    expect(wrapper).toMatchSnapshot();
});

it('should fire callback function onPress on Android', () => {
    isAndroid.mockImplementationOnce(() => true);
    const onPress = jest.fn();
    const wrapper = shallow(<Button label="Stäng" onPress={onPress} />);
    wrapper.simulate('press');
    expect(onPress).toHaveBeenCalledTimes(1);
});

it('should fire callback function onPress on iOS', () => {
    isAndroid.mockImplementationOnce(() => false);
    const onPress = jest.fn();
    const wrapper = shallow(<Button label="Stäng" onPress={onPress} />);
    wrapper.simulate('press');
    expect(onPress).toHaveBeenCalledTimes(1);
});
