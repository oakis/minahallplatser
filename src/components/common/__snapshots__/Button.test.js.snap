// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot on Android 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "borderRadius": 3,
        "elevation": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "marginBottom": 5,
        "paddingBottom": 10,
        "paddingTop": 10,
        "shadowColor": "#282828",
        "shadowOffset": Object {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0.5,
        "shadowRadius": 1,
      },
      Object {
        "backgroundColor": undefined,
      },
      Object {},
    ]
  }
>
  <MaterialIcon
    color="#fff"
    name=""
    size={24}
  />
  <Text
    style={
      Array [
        Object {
          "fontFamily": "System",
          "fontSize": 16,
        },
        Object {
          "color": "#fff",
          "fontFamily": "System",
          "fontWeight": "bold",
          "marginLeft": 0,
        },
      ]
    }
  />
</View>
`;

exports[`should match snapshot on iOS 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "stretch",
      "backgroundColor": undefined,
      "borderRadius": 3,
      "elevation": 1,
      "flexDirection": "row",
      "justifyContent": "center",
      "marginBottom": 5,
      "opacity": 1,
      "paddingBottom": 10,
      "paddingTop": 10,
      "shadowColor": "#282828",
      "shadowOffset": Object {
        "height": 0,
        "width": 0,
      },
      "shadowOpacity": 0.5,
      "shadowRadius": 1,
    }
  }
>
  <MaterialIcon
    name=""
    size={24}
  />
  <Text
    style={
      Array [
        Object {
          "fontFamily": "System",
          "fontSize": 16,
        },
        Object {
          "color": "#fff",
          "fontFamily": "System",
          "fontWeight": "bold",
          "marginLeft": 0,
        },
      ]
    }
  />
</View>
`;
